### PR TITLE
MTDSA-8287 List Charges Alignments

### PR DESCRIPTION
**Pull request by ngangaNjiraini**

### CODE REVIEW

**Review by BetaDraconis — CHANGES_REQUESTED** [review]
`INVALID_INCLUDE_STATISTICAL`, `INVALID_REQUEST`, and `INVALID_DATE_RANGE` service error mappings are missing

File: app/v1/models/response/listCharges/ListChargesResponse.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ case class ListChargesResponse[I](charges: Seq[I])
 object ListChargesResponse extends HateoasLinks {
```
[the remark this answers is not in the document]
Comment (BetaDraconis) [reply app/v1/models/response/listCharges/ListChargesResponse.scala:27]: This file is also missing a Writes test

File: app/config/DocumentationController.scala [hunk 2]
```diff
@@ -22,7 +22,8 @@ import javax.inject.{Inject, Singleton}
 import play.api.http.HttpErrorHandler
 import play.api.libs.json.Json
 import play.api.mvc.{Action, AnyContent, ControllerComponents}
-import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
```
Comment (Taiwo-AB) [inline app/config/DocumentationController.scala:25]: Can you please revert back to use previous import as this import is deprecated. Info can be seen [here](https://github.com/hmrc/bootstrap-play/blob/master/bootstrap-backend-play-26/src/main/scala/uk/gov/hmrc/play/bootstrap/package.scala#L37)

File: app/v1/connectors/ListChargesConnector.scala [hunk 3]
```diff
@@ -19,7 +19,7 @@ package v1.connectors
 import config.AppConfig
 import javax.inject.{Inject, Singleton}
 import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.http.HttpClient
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
```
Comment (Taiwo-AB) [inline app/v1/connectors/ListChargesConnector.scala:22]: Can you please revert back to use previous import as this import is deprecated. Info can be seen [here](https://github.com/hmrc/bootstrap-play/blob/master/bootstrap-backend-play-26/src/main/scala/uk/gov/hmrc/play/bootstrap/package.scala#L89)

File: app/v1/models/errors/mtdErrors.scala [hunk 4]
```diff
@@ -70,6 +70,8 @@ object NoPaymentsFoundError extends MtdError("NO_PAYMENTS_FOUND", "No payments f
 
 object NoChargesFoundError extends MtdError("NO_CHARGES_FOUND", "No charges found")
```
Comment (Taiwo-AB) [inline app/v1/models/errors/mtdErrors.scala:71]: This should be removed as it is not used.

File: app/v1/models/errors/mtdErrors.scala [hunk 5]
```diff
@@ -70,6 +70,8 @@ object NoPaymentsFoundError extends MtdError("NO_PAYMENTS_FOUND", "No payments f
...
 
+object NoTransactionsFoundError extends MtdError("NO_TRANSACTIONS_FOUND", "No transactions found")
```
Comment (Taiwo-AB) [inline app/v1/models/errors/mtdErrors.scala:73]: This should also be removed as it is not used

File: test/v1/controllers/ListChargesControllerSpec.scala [hunk 6]
```diff
@@ -87,20 +87,6 @@ class ListChargesControllerSpec extends ControllerBaseSpec
       )
```
Comment (Taiwo-AB) [inline test/v1/controllers/ListChargesControllerSpec.scala:87]: Unused import here

File: test/v1/fixtures/ListChargesFixture.scala [hunk 7]
```diff
@@ -24,7 +24,7 @@ object ListChargesFixture {
   val fullDesChargeResponse: JsValue = Json.parse(
```
Comment (Taiwo-AB) [inline test/v1/fixtures/ListChargesFixture.scala:24]: minimalDesChargeResponse and fullDesListChargesSingleResponse can be removed here as they are not currently being used

File: test/v1/fixtures/ListChargesFixture.scala [hunk 8]
```diff
@@ -56,63 +56,72 @@ object ListChargesFixture {
...
+    s"""
+       |{
+       | "charges" : [$fullChargeMtdResponse]
+       |}
+      """.stripMargin
+  )
 
-  val mtdResponse: JsValue = Json.parse(
+  val ListChargesMtdResponseWithHateoas: JsValue = Json.parse(
     s"""
```
Comment (Taiwo-AB) [inline test/v1/fixtures/ListChargesFixture.scala:125]: The s can be removed here

File: app/v1/controllers/AuthorisedController.scala [hunk 9]
```diff
@@ -21,7 +21,7 @@ import play.api.mvc._
 import uk.gov.hmrc.auth.core.Enrolment
 import uk.gov.hmrc.auth.core.authorise.Predicate
 import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
```
Comment (Taiwo-AB) [inline app/v1/controllers/AuthorisedController.scala:24]: Can you please revert back to use previous import as this import is deprecated. Info can be seen [here](https://github.com/hmrc/bootstrap-play/blob/master/bootstrap-backend-play-26/src/main/scala/uk/gov/hmrc/play/bootstrap/package.scala#L37)

**Pull request merged**